GMGoogleDrive/Public/Get-GDriveAccessToken.ps1

```diff
@@ -24,31 +36,111 @@
     https://developers.google.com/identity/protocols/OAuth2
     https://developers.google.com/identity/protocols/OAuth2InstalledApp
     https://developers.google.com/identity/protocols/OAuth2WebServer
+    https://developers.google.com/identity/protocols/oauth2/service-account
 #>
 function Get-GDriveAccessToken {
 [CmdletBinding()]
     param(
-        [Parameter(Mandatory, Position=0, ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName='ClientID', Mandatory, Position=0, ValueFromPipelineByPropertyName)]
         [string]$ClientID,
 
-        [Parameter(Mandatory, Position=1, ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName='ClientID', Mandatory, Position=1, ValueFromPipelineByPropertyName)]
         [string]$ClientSecret,
 
-        [Parameter(Mandatory, Position=2, ValueFromPipelineByPropertyName)]
-        [string]$RefreshToken
+        [Parameter(ParameterSetName='ClientID', Mandatory, Position=2, ValueFromPipelineByPropertyName)]
+        [string]$RefreshToken,
+
+        [Parameter(ParameterSetName='P12ServiceAccount', Mandatory, ValueFromPipelineByPropertyName)]
+        [Parameter(ParameterSetName='JSONServiceAccount', Mandatory, ValueFromPipelineByPropertyName)]
+        [string]$PathServiceAccountFile,
+
+        [Parameter(ParameterSetName='P12ServiceAccount', ValueFromPipelineByPropertyName)]
+        #[SecureString]$CertificatePassword = (ConvertTo-SecureString "notasecret" -AsPlainText -Force),
```

Comment:
PSScriptAnalyzer sees this `(ConvertTo-SecureString "notasecret" -AsPlainText -Force)` as a security risk.
My implementation avoids the alert from PSScriptAnalyzer.
From my perspective it is not a security risk as everbody can find the password in the [Google documentation](https://support.google.com/cloud/answer/6158849?hl=en&sjid=10155945132668630004-EU#zippy=%2Cservice-accounts)